ide-migration/server/migration/api/migration-service.mjs

```diff
@@ -108,7 +108,17 @@ export class MigrationService {
     }
 
     _buildHDIContainerName(duName, projectName) {
-        return `${duName}_${projectName}`.toUpperCase();
+        const groupAndContainerNamePattern = /[^\w_]/g;
+        const forbiddenCharactersReplacer = "_";
+        const forbiddenLeadingCharacter = "_";
+
+        let nameAccordingToConvention = `${duName}_${projectName}`.toUpperCase().replace(groupAndContainerNamePattern, forbiddenCharactersReplacer);
+
+        while (nameAccordingToConvention.substring(0, 1) == forbiddenLeadingCharacter) {
```

Comment:
Couldn't we remove this code if we just make the name pattern to be like `hdi_${duName}_${projectName}`? This way we would guarantee there is no `_` in the beginning

Reply:
OK